src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.Methods.cs

```diff
@@ -4177,6 +4177,115 @@ public void Control_Invoke_InvokeInvalidParameters_ThrowsTargetParameterCountExc
         Assert.Throws<TargetParameterCountException>(() => control.Invoke(method, Array.Empty<object>()));
     }
 
+#if DEBUG
+    // Here we will test for ThreadMethodEntry destructor will not be called.
+    // If we use AsyncWaitHandle and not close it, then we will crash to xUnit runner:
+    // The active test run was aborted. Reason: Test host process crashed : Unhandled exception. DebugAssertException: Method Debug.Fail failed with 'ThreadMethodEntry finalization hit!'
+    [WinFormsFact]
+    public void Control_Invoke_SameThread_ThreadMethodEntry_Finalizeation_Success()
```

Comment:
```suggestion
    public void Control_Invoke_SameThread_ThreadMethodEntry_Finalization_Success()
```

Same with the others.